cmake/kokkoskernels_git_info.cmake

```diff
@@ -0,0 +1,122 @@
+# https://jonathanhamberg.com/post/cmake-embedding-git-hash/
+
+find_package(Git QUIET)
```

Comment:
Do we need a particular version of git for the code below to work correctly?

Reply:
All of the git commands used are core functionality that has been in git for a long time. It would be hard to pinpoint exact minimum version, but similar implementations have worked fine in kokkos and [DARMA-tasking/vt](https://github.com/DARMA-tasking/vt/) on a variety of machines / configurations.